src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/_params.py

```diff
@@ -63,15 +63,15 @@ def load_arguments(self, _):
         c.argument('show', action='store_true', help='Display the credential information for the Batch account.')
 
     with self.argument_context('batch application set') as c:
-        c.argument('application_id', options_list=('--application-id',), help="The ID of the application.")
+        c.argument('application_name', options_list=('--application-name',), help="The name of the application.")
```

Comment:
do we want to make the breaking change on the option name? If yes, we should mention it on the history.